### PR TITLE
Only generate Integrations home page for `current`

### PR DESCRIPTION
**Pull request by ShaunLloyd**
## What changed?

- [x] Refactor addons page generation into separate files for readability
- [x] Generate Integrations home page conditionally with other addon pages

### CODE REVIEW

File: src/util/create-integrations-pages/create-integrations-home-page.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
+const path = require('path');
+const { ADDON_FRAGMENT } = require('./constants');
+
+const { wait, validateResponse } = require('./helpers');
+
+const PAGE_COMPONENT_PATH = path.resolve(
+  `./src/components/screens/AddonsHomeScreen/AddonsHomeScreen.js`
+);
+
+function fetchIntegrationsHomePage(createPage, graphql, skip = 0) {
```
Comment (kylegach) [inline src/util/create-integrations-pages/create-integrations-home-page.js:10]: Minor thing, but it appears `skip` is unnecessary.

File: src/util/create-integrations-pages/create-category-pages.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  });
+}
+
+module.exports = function createCategoryPages(createPage, graphql) {
+  console.log(`
+🗂️ Creating category pages
+`);
+
+  return fetchAddonsDetailPages(graphql).then((categoryPages) =>
+    generateCategoryPages(createPage, categoryPages)
```
Comment (kylegach) [inline src/util/create-integrations-pages/create-category-pages.js:53]: Can we move this to be part of `fetchAddonsDetailPages` (which should be renamed to `fetchAddonsCategoryPages`), to match the shape of the other files?

File: src/util/create-integrations-pages/helpers.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+const DELAY = 3000;
+/**
+ * @param {int} delay Time in milliseconds
+ * @returns A promise that will resolve in given milliseconds
+ */
+const wait = (delay = DELAY) => new Promise((resolve) => setTimeout(resolve, delay));
+
+function validateResponse(predicate) {
+  return (response) => {
```
Comment (kylegach) [inline src/util/create-integrations-pages/helpers.js:14]: Nice refactor to return a callback function!

**Review by kylegach — APPROVED** [review]
This looks great to me. You may wish to have @winkerVSbecks review, as he's more familiar with this side of the codebase.

File: src/util/create-integrations-pages/constants.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const ADDON_FRAGMENT = `
+  id: name
```
Comment (winkerVSbecks) [inline src/util/create-integrations-pages/constants.js:2]: I was going to recommend this approach: https://www.gatsbyjs.com/docs/reference/graphql-data-layer/using-graphql-fragments/

But I think you can't do that here cause it's running in Node and not a page query?

**Review by winkerVSbecks — APPROVED** [review]
Nicely done!

File: src/util/create-integrations-pages/constants.js (hunk 4 above)
Comment (ShaunLloyd) [reply src/util/create-integrations-pages/constants.js:2]: Yeah it didn't work in the gatsby-node context :(

**Pull request merged**